env0/resource_project.go

```diff
@@ -94,18 +84,16 @@ func resourceProjectUpdate(ctx context.Context, d *schema.ResourceData, meta int
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	payload := client.ProjectCreatePayload{
-		Name:        d.Get("name").(string),
-		Description: d.Get("description").(string),
+	var payload client.ProjectCreatePayload
+
+	if err := readResourceData(&payload, d); err != nil {
```

Comment:
so from now on we first read the resource and deserialized it
and before write we serialize it?

Reply:
@GiliFaroEnv0 

readResourceData extracts from resourceData and write to the interface (E.g. `var payload client.ProjectCreatePayload` in this case).

writeResourceData extracts from the interface (a struct usually) and write to the resourceData).

Note:
For most cases this works well. But in some cases manual d.Get("...") or d.Set("...") is required.
These two functions use golang reflection.